samples/template/README.rst

```diff
@@ -0,0 +1,305 @@
+| nRF52840 DK        | PCA10056 | `nrf52840dk_nrf52840`_ | ``nrf52840dk_nrf52840``      |
++--------------------+----------+------------------------+------------------------------+
+| nRF5340 DK         | PCA10095 | `nrf5340dk_nrf5340`_   | ``nrf5340dk_nrf5340_cpuapp`` |
++--------------------+----------+------------------------+------------------------------+
+
+Besides the development kit, a LoRa radio module is also needed for this sample if you build it with the LoRa or FSK configuration.
+The supported modules are as follows:
+
++------------+---------------+------------+
+| Radio chip | Board name    | Interface  |
++============+===============+============+
+| `SX1262`_  | SX126xMB2xAS  | SPI @ 8Mhz |
++------------+---------------+------------+
+
+.. note::
+   The LoRa radio module shield must be connected to the development kit header, and the antenna has to be connected to the radio module.
+   For the exact pin assignment, refer to the :ref:`setting_up_hardware_semtech_pinout` section.
+
+.. _template_overview:
+
+Overview
+********
+
+You can use this sample as a starting point to implement a Sidewalk device.
+The sample shows implementation of the Sidewalk API.
+
+.. _template_ui:
+
+User interface
+**************
+
+A button action is triggered when you release the button.
+To use a long press action, hold a button for 2 seconds or longer, and release it.
+
+The button assignment is as follows:
+
+   * Button 1 (long press):
+      Factory reset - The application informs the Sidewalk stack about the factory reset event.
+      The Sidewalk library clears its configuration from the non-volatile storage.
+      After a successful reset, the device needs to be registered with the cloud services again.
+
+   * Button 2 (short press):
+      In BLE link type - toggle Connection Request - The device requests the Sidewalk Gateway to initiate a connection while the device is advertising through Bluetooth LE.
+      After the connection is dropped, the user has to set the beacon state again.
+      Gateways may not always be able to process this request, as it depends on the number of devices connected to it.
+
+   * Button 2 (short press):
+      In LoRa or FSK link type - get current Device Profiles - The divice profile paramters are printed in logs.
+
+   * Button 2 (long press):
+      In LoRa or FSK link type - switch between Device Profiles - The application sets the device profile fields like link, profile ID, window count, sync and async intervals for communication.
+
+   * Button 3:
+      Send Hello - This action will queue a message to the cloud.
+      If Sidewalk is not ready, it will simply show an error without sending the message.
+      The queue will be processed eventually, and all the queued messages will be sent.
+
+   * Button 4 (short press):
+      Change link type. There is a Circular link type change BLE -> FSK -> LoRa -> BLE ...
```

Comment:
```suggestion
      Change link type. There is a Circular link type change BLE -> FSK -> LoRa -> BLE ->. Selected link type is saved and set after sample restart. ``Button 1 (long press)`` reset link type to initial one defined by ``CONFIG_SIDEWALK_APP_LINK_TYPE_(BLE/LORA/FSK)``
```

Note: LoRa link type does not support registration and switching to it on de-registered sample causes error ``Failed to initialize sidewalk err:-6``